Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedDataParameterDefinitionTest.cs

```diff
@@ -0,0 +1,136 @@
+using Remotion.Development.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SqlServer.Parameters;
+
+[TestFixture]
+public class SqlTableValuedDataParameterDefinitionTest
+{
+  private static IEnumerable<(object[], bool)> GetTestCollections ()
+  {
+    yield return ([1, 2, 3, 5, 8, 13, 21], true); // distinct
+    yield return ([1, 2, 3, 3, 2, 1], false); // duplicates
+    yield return ([], false); // empty
+    yield return ([], true); // empty distinct
+    yield return ([ClassWithAllDataTypes.EnumType.Value0, ClassWithAllDataTypes.EnumType.Value1, ClassWithAllDataTypes.EnumType.Value2], true); // converted
+    yield return (["Value1", "Value2", "Value1"], false);
+  }
+
+  [Test]
+  [TestCaseSource(nameof(GetTestCollections))]
+  public void GetParameterValue_ReturnsSqlTableValuedParameterValue ((object[], bool) testCase)
+  {
+    var items = testCase.Item1;
+    var isDistinct = testCase.Item2;
+
+    var storageTypeInformation = new SqlStorageTypeInformationProvider().GetStorageType(items.FirstOrDefault());
+    var convertedValues = items.Select(i => storageTypeInformation.ConvertToStorageType(i)).ToArray();
+
+    var parameterDefinition = new SqlTableValuedDataParameterDefinition(storageTypeInformation, isDistinct);
+
+    var result = parameterDefinition.GetParameterValue(items);
+
+    Assert.That(result, Is.InstanceOf<SqlTableValuedParameterValue>());
+
+    var tvpValue = result.As<SqlTableValuedParameterValue>();
+
+    if(isDistinct)
+      Assert.That(tvpValue.TableTypeName, Is.EqualTo($"TVP_{storageTypeInformation.StorageDbType}{SqlTableValuedDataParameterDefinition.DistinctCollectionTableTypeNameSuffix}"));
+    else
+      Assert.That(tvpValue.TableTypeName, Is.EqualTo($"TVP_{storageTypeInformation.StorageDbType}"));
+
+    Assert.That(tvpValue.ColumnMetaData.Count, Is.EqualTo(1));
+
+    var sqlMetaData = tvpValue.ColumnMetaData.Single();
+    Assert.That(sqlMetaData.Name, Is.EqualTo("Value"));
+
+    var conversionHelper = new SqlParameter();
+    conversionHelper.DbType = storageTypeInformation.StorageDbType;
+    var sqlDbType = conversionHelper.SqlDbType;
+    Assert.That(sqlMetaData.SqlDbType, Is.EqualTo(sqlDbType));
+
+    Assert.That(tvpValue.Select(record => record.GetValue(0)), Is.EquivalentTo(convertedValues));
+  }
+
+  [Test]
+  public void GetParameterValue_WithMismatchedItemType_ThrowsArgumentException ()
+  {
+    var items = new[] { "Value1", "Value2" };
+    var storageTypeInformation = StorageTypeInformationObjectMother.CreateIntStorageTypeInformation();
+
+    var parameterDefinition = new SqlTableValuedDataParameterDefinition(storageTypeInformation, true);
+
+    Assert.That(
+        () => parameterDefinition.GetParameterValue(items),
+        Throws.InstanceOf<ArgumentException>().With
+            .ArgumentExceptionMessageEqualTo($"Item 0 of parameter 'value' has the type '{typeof(string)}' instead of '{typeof(int)}'.", "value"));
+  }
```

Comment:
wouldn't this test be more explicit if we did a non-generic collection with items `{1, "stringValue", 2}`?